grpc/src/main/resources/assets/pt/src/view/map/Map.js

```diff
@@ -61,8 +61,11 @@ class LeafletComponent extends React.Component {
       iconSize: 10
     }));
     let bbox = this.props.info.bbox;
-    console.log(bbox)
-    this.map.fitBounds(Leaflet.latLngBounds(Leaflet.latLng(bbox[1], bbox[0]), Leaflet.latLng(bbox[3], bbox[2])));
+    if (this.props.from != null && this.props.to != null) {
+        bbox = [this.props.from.long, this.props.from.lat, this.props.to.long, this.props.to.lat];
```

Comment:
this is a total nit, but it's a little nonstandard for this component to receive its props, but then overwrite the `bbox` prop based on the value of other props. whoever is setting the props could instead set the bbox correctly with respect to from/to. looks like these props ultimately come from `App` React state and are initialized on app refresh from the query string at https://github.com/replicahq/graphhopper/blob/e3e445186965b056266d65272aea44f8f054e0a8/grpc/src/main/resources/assets/pt/src/data/Query.js#L39-L45, so we could update that query string parsing to also set the bbox based on from/to.

another option would be to drop `bbox` as a prop and move it to be in this component's state. that way this component can manage the bbox and always correctly derive it from the from/to props when the component mounts. the parent component doesn't need to track the bbox, it can just worry about from/to.

anyways looks like what you have is working fine so these approaches are definitely not necessary, and i don't know how deeply we want to invest in the router GUI generally. i just bring it up in case these more react-y patterns would make the GUI code easier to understand and maintain for the future. i have a bit of previous react experience so this stuff jumped out at me a bit, but also completely understand if polishing up GUI code just isn't worth the investment 🙂

Reply:
Really good points here! I'm the first to admit that I'm definitely not a React expert myself, so I appreciate any insights you can lend here 😁  IIRC this code was all written by @michaz , and might not follow React best practices perfectly - that's probably OK for what we need it for, but I think if we are touching the code for changes like this anyway, doing a bit of cleanup never hurts.

I refactored a bit to achieve what I _think_ we wanted here - we're actually hitting an `info` endpoint of the router to set the bbox string after we do the query parsing in `parsePoints`/`parseQuery`, so I stuck the new logic in that area. Now, we return a bbox reflecting the area around the points, if they exist; otherwise, we set the bbox based on the result of the `info` endpoint

Reply:
Nice, I like the change!

Reply:
Confirmed, I know nothing about React.